### PR TITLE
[noup] zephyr: Remove the wpa_supp_els_pkc_mbedtls_config.h

### PR DESCRIPTION
**Pull request by MaochenWang1**
Remove wpa_supp_els_pkc_mbedtls_config.h, as this header file
contains PSA_CRYPTO_DRIVER_ELS_PKC, and ELS-PKC is a proprietary
component of nxp to provides HW acceleration for psa-apis.

### CODE REVIEW

File: port/mbedtls/wpa_supp_els_pkc_mbedtls_config.h [hunk 1]
```diff
@@ -3995,16 +3995,77 @@ void *pvPortCalloc(size_t num, size_t size); /*Calloc for HEAP3.*/
  */
 
 #define MBEDTLS_ALLOW_PRIVATE_ACCESS
+#define MBEDTLS_CTR_DRBG_ENABLED
+#define MBEDTLS_CIPHER_AES_ENABLED
+#define MBEDTLS_CIPHER_MODE_CBC_ENABLED
+#define MBEDTLS_CIPHER_PADDING_PKCS7
+#define PSA_CRYPTO_DRIVER_ELS_PKC
```
Comment (krish2718) [inline port/mbedtls/wpa_supp_els_pkc_mbedtls_config.h:4002]: can you please share some references to ELS-PKC and also elaborate commit log about the nature of error and how the fix fixes it? (typo in commit log, els_pck -> ELS PKC)

Comment (MaochenWang1) [reply port/mbedtls/wpa_supp_els_pkc_mbedtls_config.h:4002]: Hi @krish2718 , what do you mean that some references to ELS-PKC?

Comment (MaochenWang1) [reply port/mbedtls/wpa_supp_els_pkc_mbedtls_config.h:4002]: update the commit message

Comment (krish2718) [reply port/mbedtls/wpa_supp_els_pkc_mbedtls_config.h:4002]: I am not familiar with ELS-PKC, so, was wondering if you can provide some context about this and how/why this is used in WPA supplicant? I only see below usage in Zephyr repo, but that still doesn't give enough context.

```
$ ag ELS_PKC ../
../modules/hal/nxp/mcux/mcux-sdk/drivers/common/fsl_common.h
201:    kStatusGroup_ELS_PKC        = 161, /*!< Group number for ELS PKC status codes. */

../modules/lib/hostap/port/mbedtls/wpa_supp_els_pkc_mbedtls_config.h
58:#define PSA_CRYPTO_DRIVER_ELS_PKC

../modules/tee/tf-m/trusted-firmware-m/platform/ext/target/nxp/common/Native_Driver/drivers/fsl_common.h
198:    kStatusGroup_ELS_PKC        = 161, /*!< Group number for ELS PKC status codes. */
```

Comment (MaochenWang1) [reply port/mbedtls/wpa_supp_els_pkc_mbedtls_config.h:4002]: Hi @krish2718 @jukkar @rlubos  please review again

Comment (krish2718) [reply port/mbedtls/wpa_supp_els_pkc_mbedtls_config.h:4002]: > Hi @krish2718 the mention of els_pkc is not appropriate in the commit message, so I remove it. actually the Els-pkc is proprietary component of nxp, which provides HW acceleration for psa-apis

Thanks. So, this header file isn't used upstream?

Comment (MaochenWang1) [reply port/mbedtls/wpa_supp_els_pkc_mbedtls_config.h:4002]: this header file isn't used upstream

Comment (krish2718) [reply port/mbedtls/wpa_supp_els_pkc_mbedtls_config.h:4002]: Thanks, then IMHO we should remove this as it's only NXP related. @jukkar WDYT?

Comment (jukkar) [reply port/mbedtls/wpa_supp_els_pkc_mbedtls_config.h:4002]: Yes, I agree

Comment (krish2718) [reply port/mbedtls/wpa_supp_els_pkc_mbedtls_config.h:4002]: @MaochenWang1 can you please update this PR to remove this file instead?

Comment (MaochenWang1) [reply port/mbedtls/wpa_supp_els_pkc_mbedtls_config.h:4002]: yes, will update

**Review by krish2718 — APPROVED** [review]

File: port/mbedtls/wpa_supp_els_pkc_mbedtls_config.h (hunk 1 above)
Comment (MaochenWang1) [reply port/mbedtls/wpa_supp_els_pkc_mbedtls_config.h]: Hi @jukkar @krish2718  remove as your comment

**Review by jukkar — APPROVED** [review]

**Pull request merged**